File: frontend/src/types/ai.ts
export interface AIRecommendation {
    title: string;
    reason: string;
    difficulty: string;
    estimatedMinutes: number;
    focus: string;
}

export interface AIFeedback {
    summary: string;
    strengths: string[];
    weaknesses: string[];
    suggestions: string[];
    nextStep: string;
    encouragement: string;
}

export interface AIFeedbackRequest {
    userId: string;
    practiceId: string;
    score: number;
    correct: number;
    wrong: number;
    userAnswer?: string;
}

File: backend/internal/dto/ai.go
package dto

type AIRecommendationResponse struct {
	Title            string `json:"title"`
	Reason           string `json:"reason"`
	Difficulty       string `json:"difficulty"`
	EstimatedMinutes int    `json:"estimatedMinutes"`
	Focus            string `json:"focus"`
}

type AIFeedbackRequest struct {
	UserID     string `json:"userId"`
	PracticeID string `json:"practiceId"`
	Score      int    `json:"score"`
	Correct    int    `json:"correct"`
	Wrong      int    `json:"wrong"`
	UserAnswer string `json:"userAnswer"`
}

type AIFeedbackResponse struct {
	Summary       string   `json:"summary"`
	Strengths     []string `json:"strengths"`
	Weaknesses    []string `json:"weaknesses"`
	Suggestions   []string `json:"suggestions"`
	NextStep      string   `json:"nextStep"`
	Encouragement string   `json:"encouragement"`
}